### PR TITLE
Auto package

### PR DESCRIPTION
**Pull request by sigvef**

### CODE REVIEW

File: nin/backend/compile.js [hunk 1]
```diff
@@ -32,18 +33,44 @@ function renderError() {
...
+function slugifyTitle(title) {
+  return slugify(title).toLowerCase();
+}
+
+async function createArchive(projectPath) {
+  return new Promise(resolve => {
+    process.stdout.write(chalk.yellow('\nCreating zip archive'));
+    const metadata = utils.getProjectMetadata(projectPath);
+    const slug = slugifyTitle(metadata.projectSettings.title);
+    const outputStream = fs.createWriteStream('bin/' + slug + '.zip');
```
Comment (aleksanb) [inline nin/backend/compile.js]: Here and other  places you create paths you should use p.join to get the correct path separators for windows and GNU/linux.

File: nin/backend/compile.js [hunk 2]
```diff
@@ -32,18 +33,44 @@ function renderError() {
...
+    const archive = archiver('zip');
+    archive.pipe(outputStream);
+    let foundErrors = false;
+    archive.on('warning', e => {
```
Comment (aleksanb) [inline nin/backend/compile.js]: Alternatively reject the promise with the error message here,
and move error handling to a single place further out in the code.

File: nin/backend/compile.js [hunk 3]
```diff
@@ -83,12 +110,14 @@ async function res(projectPath) {
 }
 
 async function collect(projectPath, data) {
-  process.stdout.write(chalk.yellow('\nGenerating demo.html'));
+  const projectMetadata = utils.getProjectMetadata(projectPath);
+  const slug = slugifyTitle(projectMetadata.projectSettings.title);
+  process.stdout.write(chalk.yellow('\nGenerating ' + slug + '.html'));
```
Comment (aleksanb) [inline nin/backend/compile.js]: No need to get the slug separately here, you can just `projectSettings.title` below the destructuring.

File: nin/backend/compile.js (hunk 2 above)
Comment (sigvef) [inline nin/backend/compile.js]: Either way is fine. The way it is done now is consistent with the rest of the error handling in this file, so I think it is best to leave it as-is for now.

File: nin/backend/compile.js (hunk 3 above)
Comment (sigvef) [inline nin/backend/compile.js]: What do you mean?

File: nin/backend/compile.js (hunk 1 above)
Comment (sigvef) [inline nin/backend/compile.js]: fixed

**Comment by sigvef** [comment]
ping

**Pull request merged**